x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/utils/NativeMemoryCalculator.java

```diff
@@ -70,12 +73,56 @@ private static OptionalLong allowedBytesForMl(String nodeBytes, String jvmBytes,
         return OptionalLong.of(allowedBytesForMl(machineMemory, jvmMemory, maxMemoryPercent, useAuto));
     }
 
-    public static int modelMemoryPercent(long machineMemory, long jvmSize, int maxMemoryPercent, boolean useAuto) {
+    public static long calculateApproxNecessaryNodeSize(long nativeMachineMemory, Long jvmSize, int maxMemoryPercent, boolean useAuto) {
+        if (nativeMachineMemory == 0) {
+            return 0;
+        }
         if (useAuto) {
-            // It is conceivable that there is a machine smaller than 200MB.
-            // If the administrator wants to use the auto configuration, the node should be larger.
+            // TODO utilize official ergonomic JVM size calculations when available.
+            final long unboxedJvm;
```

Comment:
There would be more symmetry in the code and it might be easier to reason about whats going on if you rename the existing `dynamicallyCalculateJvm` method to `dynamicallyCalculateJvmSizeFromNodeSize` and introduce a new method `dynamicallyCalculateJvmSizeFromMlNativeMemory` that does the bit below.  Then this line can become:

```
jvmSize = jvmSize == null ? dynamicallyCalculateJvmSizeFromMlNativeMemory(nativeMachineMemory) : jvmSize;
```

which matches the pattern that's been used elsewhere, and the names of the two utility methods make clear why they're different: one is taking total node memory as input while the other takes ML native memory requirement as input.